woocommerce/woocommerce-android
# Survey type inbox note

Closes: #6144 


### Description
Add support for survey type inbox notes. Inbox notes should have a different design than the rest of the notes. With outlined "Yes" and "No" buttons. These buttons will simply send the actiondId to the API and then we the note will be marked as "Actioned". Once Inbox note is actioned a "thank you" message will be displayed and buttons will be hidden. 

### Testing instructions

This Inbox feature is hidden under a feature flag for debug builds only, so the generated apk in this PR won't display the "Inbox" option inside More Menu. You'll need to checkout this branch and compile the app to test it. Sorry about that.

- Open More Menu -> Inbox
- Scroll until you find a survey type inbox. I recommend checking in `woo-core` notes first to see if you have any note of type survey for the given site. This notes have a Yes and No button.  If you have never interacted/dismissed the notes there should be at least 2. 
- Click on Yes or No button and check that the note gets updated accordingly. 

Disclaimer: Once I have all the essential task for Inbox done I'll take a look into animating this interaction as well as dismissing note interaction. As of today, animating removal of elements inside `LazyColum` is not straight forward. 

### Images/gif

https://user-images.githubusercontent.com/2663464/162921016-ffb1112b-d063-485b-a410-bce823ac942b.mp4
